airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/CdcAcceptanceTests.java

```diff
@@ -251,15 +306,161 @@ public void testDeleteRecordCdcSync() throws Exception {
     waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead2.getJob());
     LOGGER.info("state after sync 2: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
 
-    assertDestinationMatches(expectedDestRecordMatchers);
+    assertDestinationMatches(ID_AND_NAME_TABLE, expectedIdAndNameRecords);
+  }
+
+  @Test
+  public void testPartialResetFromSchemaUpdate() throws Exception {
+    LOGGER.info("Starting partial reset cdc test");
+
+    final UUID connectionId = createCdcConnection();
+    LOGGER.info("Starting sync 1");
+
+    final JobInfoRead connectionSyncRead1 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead1.getJob());
+
+    final Database source = testHarness.getSourceDatabase();
+
+    List<DestinationCdcRecordMatcher> expectedIdAndNameRecords = getCdcRecordMatchersFromSource(source, ID_AND_NAME_TABLE);
+    assertDestinationMatches(ID_AND_NAME_TABLE, expectedIdAndNameRecords);
+
+    List<DestinationCdcRecordMatcher> expectedColorPaletteRecords = getCdcRecordMatchersFromSource(source, COLOR_PALETTE_TABLE);
+    assertDestinationMatches(COLOR_PALETTE_TABLE, expectedColorPaletteRecords);
+
+    StreamDescriptor idAndNameStreamDescriptor = new StreamDescriptor().namespace(SCHEMA_NAME).name(ID_AND_NAME_TABLE);
+    StreamDescriptor colorPaletteStreamDescriptor = new StreamDescriptor().namespace(SCHEMA_NAME).name(COLOR_PALETTE_TABLE);
+    assertGlobalStateContainsStreams(connectionId, List.of(idAndNameStreamDescriptor, colorPaletteStreamDescriptor));
+
+    LOGGER.info("Removing color palette table");
+    source.query(ctx -> ctx.dropTable(COLOR_PALETTE_TABLE).execute());
+
+    LOGGER.info("Refreshing schema and updating connection");
+    final ConnectionRead connectionRead = apiClient.getConnectionApi().getConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    UUID sourceId = createCdcSource().getSourceId();
+    AirbyteCatalog refreshedCatalog = testHarness.discoverSourceSchema(sourceId);
+    LOGGER.info("Refreshed catalog: {}", refreshedCatalog);
+    WebBackendConnectionUpdate update = getUpdateInput(connectionRead, refreshedCatalog, operationRead);
+    webBackendApi.webBackendUpdateConnectionNew(update);
+
+    LOGGER.info("Waiting for sync job after update to complete");
+    JobRead syncFromTheUpdate = testHarness.waitUntilTheNextJobIsStarted(connectionId);
+    waitForSuccessfulJob(apiClient.getJobsApi(), syncFromTheUpdate);
+
+    // We do not check that the source and the dest are in sync here because removing a stream doesn't
+    // delete its data in the destination
+    assertGlobalStateContainsStreams(connectionId, List.of(idAndNameStreamDescriptor));
+  }
+
+  @Test
+  public void testPartialResetFromStreamSelection() throws Exception {
+    LOGGER.info("Starting partial reset cdc test");
```

Comment:
Nit: you can pass a TestInfo test param to access the test name and log it.

Reply:
Done